contracts/bridges/BaseBridgeReceiver.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.15;
+
+import "../ITimelock.sol";
+
+contract BaseBridgeReceiver {
+    error AlreadyInitialized();
+    error BadData();
+    error Unauthorized();
+
+    event Initialized(address indexed l2Timelock, address indexed mainnetTimelock);
+    event NewL2Timelock(address indexed oldL2Timelock, address indexed newL2Timelock);
+    event NewMainnetTimelock(address indexed oldMainnetTimelock, address indexed newMainnetTimelock);
+    event ProcessMessage(address indexed messageSender, address[] targets, uint256[] values, string[] signatures, bytes[] calldatas, uint eta);
+
+    address public initializer;
+    address public mainnetTimelock;
+    address public l2Timelock;
+
+    constructor(address _initializer) {
+        initializer = _initializer;
+    }
+
+    function initialize(address _mainnetTimelock, address _l2Timelock) external {
```

Comment:
The deploy flow for the BridgeReceiver/L2 Timelock goes like this:

- deploy the BridgeReceiver with an EOA as initializer
- deploy the Timelock with BridgeReceiver address as admin
- initializer EOA calls BridgerReceiver.initialize with the mainnetTimelock and l2Timelock addresses

at that point, BridgeReceiver.initializer is set to the zero address, so there is no action that the initializer EOA can take on the BridgeReceiver contract

Reply:
The other options we considered were:

1. making `initialize` a function that could only be called by the `l2Timelock`; the major downside is that this would require initialization to be queued as a transaction, so getting the contracts initialized would take 2 days (or whatever the minimum delay is)

2. making `initialize` a public function that could be called by anyone. it would be difficult to do anything malicious with this function (calling `l2timelock.acceptAdmin` would simply revert if the timelock hadn't set the BridgeReceiver as the pendingAdmin), but it felt weird to leave that function callable by anyone

Reply:
We shouldn't necessarily be afraid of public `initialize`

Reply:
I've come to like the initializer more, the more I think about it.

it's a role with a very limited lifespan (as soon as you initialize, you are locked out of the role forever). and it prevents someone from griefing a newly-deployed BridgeReceiver contract (an unlikely event, but one that would take a governance proposal to undo)

Reply:
going to remove `initializer`

Reply:
removed